tests/test_transform.py

```diff
@@ -183,16 +183,45 @@ def makeCoeffs(nIn, nOut):
         coeffOffset = baseCoeff * out_ind
         for in_ind in range(nIn):
             coeff = baseCoeff * (in_ind + 1) + coeffOffset
-            coeffArr = [coeff, out_ind + 1] + [1 if i == in_ind else 0 for i in range(nIn)]
+            coeffArr = [coeff, out_ind + 1] + [2 if i == in_ind else 0 for i in range(nIn)]
             forwardCoeffs.append(coeffArr)
     return np.array(forwardCoeffs, dtype=float)
 
 
+def makeJacobian(nIn, nOut, inArray):
+    """Make a Jacobian matrix for the equation described by makeCoeffs.
+
+    Parameters
+    ----------
+    nIn, nOut : integers
+        the dimensions of the input and output data; see makeCoeffs
+    inArray : ndarray
+        an array of size `nIn` representing the point at which the Jacobian
+        is measured
+
+    Returns
+    ----------
+    J : numpy.ndarray
+        an nOut x nIn array of first derivatives
+    """
+    baseCoeff = 2.0 * 0.001
+    coeffs = np.empty((nOut, nIn))
+    for iOut in range(nOut):
+        coeffOffset = baseCoeff * iOut
+        for iIn in range(nIn):
+            coeffs[iOut, iIn] = baseCoeff * (iIn + 1) + coeffOffset
+            coeffs[iOut, iIn] *= inArray[iIn]
+    assert coeffs.ndim == 2
+    assert coeffs.shape == (nOut, nIn)
+    # Avoid spurious errors when comparing to a simplified array
+    return coeffs.squeeze()
```

Comment:
I think it will be less confusing having `makeJacobian` return the expected 2-dimensional array. The fact that the pybind11 wrapper may return something else for the Jacobian is a misfeature (one I hope we can eventually figure out a workaround for) that we need not emulate here. Instead consider putting the `squeeze` where the two matrices are compared.